WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/breakpoint/BreakpointCreator.java

```diff
@@ -0,0 +1,43 @@
+package nl.tudelft.watchdog.intellij.logic.breakpoint;
+
+import com.intellij.xdebugger.XExpression;
+import com.intellij.xdebugger.breakpoints.XBreakpoint;
+import nl.tudelft.watchdog.core.logic.breakpoint.Breakpoint;
+import org.jetbrains.java.debugger.breakpoints.properties.JavaBreakpointProperties;
+
+/**
+ * A factory for creating {@link Breakpoint}s from a supplied
+ * {@link XBreakpoint}.
+ */
+public class BreakpointCreator {
+
+    /**
+     * Factory method that creates and returns a {@link Breakpoint} from a given
+     * {@link XBreakpoint}.
+     */
+    public static Breakpoint createBreakpoint(XBreakpoint breakpoint) {
+        Breakpoint result = new Breakpoint(breakpoint.hashCode(),
+                BreakpointClassifier.classify(breakpoint));
+
+        // Initialize enabled and SuspendPolicy fields.
+        result.setEnabled(breakpoint.isEnabled());
+        result.setSuspendPolicy(breakpoint.getSuspendPolicy().ordinal());
+
+        // Initialize hit count field.
+        result.setHitCount(-1);
```

Comment:
As -1 indicates that the hit count is disabled in Eclipse, which should be the default value. So, for consistency the hit count is also initialized to -1 to indicate a disabled hit count in IntelliJ.